sdk/servicebus/service-bus/samples/v7/typescript/tsconfig.json

```diff
@@ -1,17 +1,23 @@
 {
   "compilerOptions": {
-    "target": "ES2020",
+    "target": "ES2023",
     "module": "commonjs",
-    "moduleResolution": "node",
-    "resolveJsonModule": true,
-    "esModuleInterop": true,
-    "allowSyntheticDefaultImports": true,
+    "lib": [],
+    "importHelpers": true,
     "strict": true,
     "alwaysStrict": true,
-    "outDir": "dist",
-    "rootDir": "src"
+    "noUnusedLocals": true,
```

Comment:
I feel that this is recommending options used for our development to the sample consumers.  If there's a good set we want to recommend, why not have them as static?  Keeping the config simple may be beneficial too so consumers can focus on the sample code.

Is the goal here to enable type-checking transformed/published code from samples-dev?

Reply:
These options are currently what we use to typecheck our samples. Whatever options we think should be used to typecheck samples should be in the `tsconfig.samples.json`.

Reply:
> Whatever options we think should be used to typecheck samples should be in the `tsconfig.samples.json`.

we currently only type-check `samples-dev`, not the published version

Reply:
I see, if a different config is used, there may be unexpected behaviors.

Reply:
The point of this PR is that there is no good reason to make a distinction between `samples-dev` and `samples/<version>/typescript/src`. They are the exact same code and should be typechecked in the same way.